vignettes/data-resource.Rmd

```diff
@@ -0,0 +1,205 @@
+---
+title: "Data Resource"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Data Resource}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+[Data Resource](https://specs.frictionlessdata.io/data-resource/) (further referred to as "resource") is a simple format to describe a data resource such as a individual table or file, including its name, format, path, etc.
+
+## Support
+
+### Reading
+
+frictionless can read a Data Package with resources of any type. You can list those with `resources()`:
+
+```{r}
+library(frictionless)
+x <- example_package
+
+# List resources
+resources(x)
+```
+
+Reading the data from a resource with `read_resource()` is limited to [Tabular Data Resources](https://specs.frictionlessdata.io/tabular-data-resource/):
+
+```{r}
+# Read (tabular) resource
+read_resource(x, "deployments")
+```
+
+### Mutating
+
+You can remove any of those resources with `remove_resource()`:
+
+```{r}
+# Remove resource (but don't assign to x)
+remove_resource(x, "deployments")
+```
+
+When creating or replacing a resource, `add_resource()` only supports Tabular Data Resources. The provided data must therefore be a data frame or a tabular file (e.g. CSV):
+
+```{r}
+# Add (tabular) resource from a data frame (but don't assign to x)
+add_resource(x, "iris", data = iris)
+
+# Replace a (tabular) resource from a file (but don't assign to x)
+path <- system.file("extdata", "deployments.csv", package = "frictionless")
+add_resource(x, "deployments", data = path, replace = TRUE)
+```
+
+### Writing
+
+`write_package()` will leave originally included resources as is and write newly added resources as Tabular Data Resources.
+
+## Properties
+
+The following sections describe how frictionless supports the Data Package properties.
+
+### name
+
+[`name`](https://specs.frictionlessdata.io/data-resource/#name) is required. It is used to identify a resource in `read_resource()`, `add_resource()` and `remove_resource()` (always as the second argument):
+
+```{r}
+deployments <- read_resource(x, resource_name = "deployments")
+```
+
+When creating a resource, `add_resource()` will set `name` to the provided `resource_name`.
+
+### path or data
+
+[`path`](https://specs.frictionlessdata.io/data-resource/#data-location) or `data` is required. Providing both is not allowed.
+
+#### path
+
+`path` is for data in files (e.g. a CSV file). It can be a local path or URL, as long as it resolves. Supported protocols are `http`, `https`, `ftp`, `sftp` and `sftp`. Absolute paths (`/`) or relative parent paths (`../`) are not allowed to avoid security vulnerabilities.
```

Comment:
Maybe `resolve` is the wrong verb here, but I mean a local path or URL that points to a file (i.e. it does not return file not found or a 404). I browsed around, but don't immediately have an alternative. Suggestions?

Reply:
It is specified in the first sentence it must be a file. So I sugegst to keep it simple and just remove that part.

>  `path` is for data in files (e.g. a CSV file). It can be a local path or URL. Supported protocols are `http`, `https`, `ftp`, `sftp` and `sftp`. Absolute paths (`/`) or relative parent paths (`../`) are not allowed to avoid security vulnerabilities.